File: java/kotlin-in-action/src/main/kotlin/com/ch03/java/StringUtil.java
package com.ch03.java;

import com.ch03.strings.*;

public class StringUtil {

    public static void main(String[] args) {
        char letter1 = StringUtilKt.lastChar("Java");
        char letter2 = StringUtilKt.getLastChar("Java");
        System.out.println(letter1);
        System.out.println(letter2);
    }
}

File: java/kotlin-in-action/src/main/kotlin/com/ch03/strings/StringUtil.kt
package com.ch03.strings

fun String.lastChar(): Char = this[length - 1]

val String.lastChar: Char
    get() = get(length -1 )

var StringBuilder.lastChar: Char
    get() = get(length - 1)
    set(value: Char) = this.setCharAt(length - 1, value)